plumbing/protocol/packp/updreq_decode.go

```diff
@@ -76,14 +77,7 @@ func errMalformedCommand(err error) error {
 
 // Decode reads the next update-request message form the reader and wr
 func (req *ReferenceUpdateRequest) Decode(r io.Reader) error {
-	var rc io.ReadCloser
-	var ok bool
-	rc, ok = r.(io.ReadCloser)
-	if !ok {
-		rc = io.NopCloser(r)
-	}
-
-	d := &updReqDecoder{r: rc, s: pktline.NewScanner(r)}
+	d := &updReqDecoder{r: io.NopCloser(r), s: pktline.NewScanner(r)}
```

Comment:
This is trickier, as in the existing behaviour the callee is responsible for closing the reader `r`, with this change this swifts to the caller. For breaking changes such as this, the commit/PR would have to target `v6-exp` instead.